### PR TITLE
[REVIEW] Adding continuous_ucx_progress() to the public API

### PR DESCRIPTION
**Pull request by madsbk**

### CODE REVIEW

File: ucp/_libs/core.pyx [hunk 1]
```diff
@@ -515,11 +515,12 @@ cdef class ApplicationContext:
             event_loop.create_task(_non_blocking_mode(weakref.ref(self)))
         )
 
-    def continually_ucx_prograss(self):
+    def continually_ucx_prograss(self, event_loop=None):
```
Comment (pentschev) [inline ucp/_libs/core.pyx]: Is it supposed to be progress maybe? I think a better name would be `continuous_ucx_progress`.

Comment (madsbk) [reply ucp/_libs/core.pyx]: Right :)

File: ucp/_libs/core.pyx [hunk 2]
```diff
@@ -515,11 +515,12 @@ cdef class ApplicationContext:
             event_loop.create_task(_non_blocking_mode(weakref.ref(self)))
         )
 
-    def continually_ucx_prograss(self):
-        """Guaranties continually UCX prograss"""
-        loop = asyncio.get_event_loop()
+    def continuous_ucx_progress(self, event_loop=None):
+        """Guaranties continually UCX progress"""
```
Comment (pentschev) [inline ucp/_libs/core.pyx]:
```suggestion
        """Guarantees continuous UCX progress"""
```

File: ucp/public_api.py [hunk 3]
```diff
@@ -137,6 +137,25 @@ def progress():
     return _get_ctx().progress()
 
 
+def continuous_ucx_progress(event_loop=None):
+    """Guaranties continually UCX progress
```
Comment (pentschev) [inline ucp/public_api.py]:
```suggestion
    """Guarantees continuous UCX progress
```

File: ucp/public_api.py [hunk 4]
```diff
@@ -137,6 +137,25 @@ def progress():
...
+
+    Use this function to associate UCX progress with an even loop.
```
Comment (pentschev) [inline ucp/public_api.py]:
```suggestion
    Use this function to associate UCX progress with an event loop.
```

File: ucp/public_api.py [hunk 5]
```diff
@@ -137,6 +137,25 @@ def progress():
...
+    Notice, multiple even loops can be associate with UCX progress.
```
Comment (pentschev) [inline ucp/public_api.py]:
```suggestion
    Notice, multiple event loops can be associate with UCX progress.
```

File: ucp/public_api.py [hunk 6]
```diff
@@ -137,6 +137,25 @@ def progress():
...
+
+    This function is automatically called when calling
+    `create_listener()` or `create_endpoint()`.
+
+    Parameters
+    ----------
+    event_loop: asyncio.event_loop, optional
+        The even loop to evoke UCX progress. If None,
```
Comment (pentschev) [inline ucp/public_api.py]:
```suggestion
        The event loop to evoke UCX progress. If None,
```

**Comment by pentschev** [comment]
rerun tests

File: ucp/public_api.py [hunk 7]
```diff
@@ -137,6 +137,25 @@ def progress():
     return _get_ctx().progress()
 
 
+def continuous_ucx_progress(event_loop=None):
```
Comment (jakirkham) [inline ucp/public_api.py]: Is this variable suppose to be used?

File: ucp/public_api.py [hunk 8]
```diff
@@ -137,6 +137,25 @@ def progress():
...
+    `create_listener()` or `create_endpoint()`.
+
+    Parameters
+    ----------
+    event_loop: asyncio.event_loop, optional
+        The event loop to evoke UCX progress. If None,
+        `asyncio.get_event_loop()` is used.
+    """
+
+    _get_ctx().continuous_ucx_progress()
```
Comment (jakirkham) [inline ucp/public_api.py]:
```suggestion
    _get_ctx().continuous_ucx_progress(event_loop= event_loop)
```

File: ucp/public_api.py (hunk 7 above)
Comment (jakirkham) [reply ucp/public_api.py]: Added a suggested fix below.

**Comment by quasiben** [comment]
rerun tests

**Comment by madsbk** [comment]
@pentschev and @jakirkham, thanks for the suggestions

**Pull request merged**